api/src/main/java/net/labymod/addons/waypoints/waypoint/WaypointMeta.java

```diff
@@ -1,57 +1,61 @@
 package net.labymod.addons.waypoints.waypoint;
 
+import net.labymod.addons.waypoints.Waypoints;
 import net.labymod.api.client.component.Component;
 import net.labymod.api.util.Color;
 import net.labymod.api.util.math.vector.FloatVector3;
 import org.jetbrains.annotations.Nullable;
 
 public class WaypointMeta {
 
+  @Nullable
+  private final String world;
+  private final String server;
+  private final String dimension;
   private Component title;
   private Color color;
   private WaypointType type;
   private FloatVector3 location;
   private boolean visible;
-  @Nullable
-  private String world;
 
 
   public WaypointMeta(Component title, Color color, WaypointType type, FloatVector3 location,
-      boolean visible,
-      @Nullable String world) {
+      boolean visible, @Nullable String world, String server, String dimension) {
     this.title = title;
     this.color = color;
     this.type = type;
     this.location = location;
     this.visible = visible;
     this.world = world;
+    this.server = server;
+    this.dimension = dimension;
   }
 
-  public Component getTitle() {
+  public Component title() {
     return this.title;
   }
 
   public void setTitle(Component title) {
     this.title = title;
   }
 
-  public Color getColor() {
+  public Color color() {
     return this.color;
   }
 
   public void setColor(Color color) {
     this.color = color;
   }
 
-  public WaypointType getType() {
+  public WaypointType type() {
     return this.type;
   }
 
   public void setType(WaypointType type) {
     this.type = type;
   }
 
-  public FloatVector3 getLocation() {
+  public FloatVector3 location() {
     return this.location;
   }
```

Comment:
Please don't just remove/rename any methods in the api, for example the Minimap needs them.